### PR TITLE
Pull request for pi-touchscreen-timeout - CPU usage modification and command line args

### PR DESCRIPTION
**Pull request by JoeHartley3**
Tim, I've made some changes to your timeout.c program to support the timeout period as a command line argument, and also to support the possibility of a touchscreen not at /dev/input/event0.  The latter solved a problem I had with using a handheld barcode scanner with the RPi which bumped the touchscreen to event1.  I've also included a shell script to start the timeout program at boot time, which automatically determines the touchscreen device.

Of more importance is the use of nanosleep in the main loop to bring the CPU usage down from 100% on a core to around 1%.  I hope you find these useful.  Thanks for the initial work!

Joe Hartley

### CODE REVIEW

File: timeout.c [hunk 1]
```diff
@@ -1,69 +1,116 @@
...
+                        }
+                }
+
+                if(difftime(now, touch) > timeout){
+                        if(on == '0'){
+                                printf("Turning Off\n");
+                                on = '1';
+                                write(lightfd, &on, sizeof(char));
+                        }
+                nanosleep(&sleepTime, NULL);
```
Comment (timothyhollabaugh) [inline timeout.c]: Should the sleep be outside of this if statement and directly inside the main loop?

File: timeout.c [hunk 2]
```diff
@@ -1,69 +1,116 @@
...
-		if(difftime(now, touch) > timeout){
-			if(on == '0'){
-				printf("Turning Off\n");
-				on = '1';
-				write(lightfd, &on, sizeof(char));
-			}
-		}
-	}
+        if (argc != 3) {
+                printf("Usage: touch-blanker <timeout_sec> <device>\n");
```
Comment (timothyhollabaugh) [inline timeout.c]: Should `touch-blanker` be `timeout` to be consistent with the binary name?

**Review by timothyhollabaugh — COMMENTED** [review]
This looks pretty good. I do not have a pi touchscreen with me to test, but it looks like it should work.

**Comment by JoeHartley3** [comment]
On Sat, 21 Apr 2018 08:10:00 -0700
Tim <notifications@github.com> wrote:

> timothyhollabaugh commented on this pull request.

> Should the sleep be outside of this if statement and directly inside the main loop?

Indeed it should.  That's what I get for going back to C after years of Python.

> Should `touch-blanker` be `timeout` to be consistent with the binary name?

Yes, it should.  Because there's already a /bin/timeout in Raspbian, I 
use a different executable name and missed this.

Thanks!

-- 
======================================================================
       Joe Hartley - UNIX/network Consultant - jh@brainiac.com
 Without deviation from the norm, "progress" is not possible. - FZappa

**Comment by timothyhollabaugh** [comment]
Would you like to make these changes before I merge?

**Comment by JoeHartley3** [comment]
On Sat, 21 Apr 2018 11:51:17 -0700
Tim <notifications@github.com> wrote:

> Would you like to make these changes before I merge?

I would, but won't be able to get to it until Monday.

-- 
======================================================================
       Joe Hartley - UNIX/network Consultant - jh@brainiac.com
 Without deviation from the norm, "progress" is not possible. - FZappa

**Comment by JoeHartley3** [comment]
On Sat, 21 Apr 2018 11:51:17 -0700
Tim <notifications@github.com> wrote:

> Would you like to make these changes before I merge?

Tim, I've updated my repo, the changes should merge into the same pull request,
let me know if you don't see them.

Thanks again!

-- 
======================================================================
       Joe Hartley - UNIX/network Consultant - jh@brainiac.com
 Without deviation from the norm, "progress" is not possible. - FZappa

**Pull request merged**